src/OpenTelemetry/Trace/Internal/FanOutActivityProcessor.cs

```diff
@@ -41,7 +41,7 @@ public BroadcastActivityProcessor(IEnumerable<ActivityProcessor> processors)
                 throw new ArgumentException($"{nameof(processors)} collection is empty");
             }
 
-            this.processors = processors;
+            this.processors = new List<ActivityProcessor>(processors);
```

Comment:
Should we cast instead of creating a `new`. Earlier implementation of using `IEnumerable` was good. Is there a reason to change this to list?

Reply:
Issue #980 points to a perf issue

Reply:
My guess would be it was changed so that if the user changes it after the call, it won't impact us. But I like it because it will also save an allocation on every Activity (when doing the forearch in OnStart & OnEnd).

Reply:
@rajkumar-rangaraj refer to this https://github.com/open-telemetry/opentelemetry-dotnet/pull/735#discussion_r441634805